### PR TITLE
fix: Use exception response_handler in User Delete Exception, instead …

### PR DESCRIPTION
**Pull request by dannyjgibson**
…of a not found response (404)

# Description

## JIRA ticket

[SE-228]

## Summary

I am unable to DELETE Users via the Authserver API. I can GET the same User I am trying to DELETE via the same API with the same Auth Token, but a DELETE returns a 404:
```
{"status": "Error", "code": 404, "messages": ["No resource with identifier '95afbef10d584472bad5ad41f726e83b' found."]}
```

The change in this PR should return a more precise message about what is preventing the DELETE to work.

[SE-228]: https://brighthiveio.atlassian.net/browse/SE-228

### CODE REVIEW

**Review by reginafcompton — CHANGES_REQUESTED** [review]
@dannyjgibson - I think this adjustment makes sense, in general. However, I am not sure you'll get the response you need for debugging. The `exception_response` either returns "Unknown" or an "IntegrityError": https://github.com/brighthive/authserver/blob/fc647a67cf9f3853874f0589f538787594bea57d/authserver/utilities/responses.py#L155-L158 

With SE-228, we do not know the nature of the exception. So...you have a couple options:

1. You can log the error, and then, check with John to get the trace. I suggest registering an error handler with authserver, and including the logs there. You can use DTM API as a reference: https://github.com/brighthive/data-trust-manager-api/blob/6e54e2935e161464953b4277b73024229a711cf9/data_trust_manager/app/app.py#L70
2. You can try to extend the `exception_response` function to return more err data. This might be a slippery slope, however – since you want to minimize what error data gets returned to the end user.

I am happy to pair on a solution, if that helps. It's definitely odd that the ORM can query users, but cannot delete them!

**Comment by dannyjgibson** [comment]
@reginafcompton Anddddd... that broke the build. I'm going to try for a bit to get the tests passing.

File: authserver/app/app.py [hunk 1]
```diff
@@ -79,6 +80,23 @@ def after_request(response):
             logging.info(jsonstr)
         return response
 
+    def handle_errors(e):
+        print(f"""{e}, app.py, line 83""")
```
Comment (reginafcompton) [inline authserver/app/app.py:84]: I suggest replacing the `print` with `logging`. (I think the print statements in the DTM API are left over from the neo4j connectivity debugging extravaganza.)

File: authserver/api/user.py [hunk 2]
```diff
@@ -146,8 +146,8 @@ def delete(self, id: str = None):
                 return self.response_handler.successful_delete_response('User', id, user_obj)
             else:
                 return self.response_handler.not_found_response(id)
-        except Exception:
-            return self.response_handler.not_found_response(id)
+        except Exception as e:
+            return self.response_handler.exception_response(exception_name=type(e).__name__)
```
Comment (reginafcompton) [inline authserver/api/user.py:150]: You have a couple options:

1. remove the try-except entirely. The error will bubble up to `handle_errors`.
2. catch the exception, and then raise a custom error (which bubbles up to `handle_errors`).

Personally, I'd go with option 1, for now.

**Review by reginafcompton — COMMENTED** [review]
A couple more changes needed!

**Pull request merged**